collectors/statsd.plugin/statsd.c

```diff
@@ -1219,7 +1219,7 @@ static int statsd_readfile(const char *filename, STATSD_APP *app, STATSD_APP_CHA
                             netdata_configured_stock_config_dir,
                             strlen(netdata_configured_stock_config_dir))) {
                         char tmpfilename[FILENAME_MAX + 1];
-                        strcpy(tmpfilename, filename);
+                        strncpyz(tmpfilename, filename, FILENAME_MAX);
```

Comment:
this is little bit out of topic but I worry we might be hurting ourselves here by using custom netdata function.
Look at the glibc implementation of strncpy, it will use memcpy. Then go to look at memcpy and you discover big ifdef hell using special instructions on different CPU depending on architecture and instruction sets available.
I think we are basically deoptimizing the system provided function. If we want to ensure string is NULL terminated even in case of dest being too short we should rewrite `strncpyz` to call `strncpy` and just always writing 0 to the last byte of DEST.

Reply:
That being said in general in netdata we have some funky historical stuff regarding string handling.